semgrep/semgrep/config_resolver.py

```diff
@@ -204,6 +210,118 @@ def _validate(  # type: ignore[misc]
+                    "application/x-yaml",
+                    "text/x-yaml",
+                    "text/yaml",
+                    "text/vnd.yaml",
+                ]
+                if content_type and any((ct in content_type for ct in yaml_types)):
+                    return parse_config_string(
+                        "remote-url",
+                        r.content.decode("utf-8", errors="replace"),
+                        filename=f"{config_url[:20]}...",
+                    )
+                else:
+                    raise SemgrepError(
+                        f"unknown content-type: {content_type} returned by config url: {config_url}. Can not parse"
+                    )
+            else:
+                raise SemgrepError(
+                    f"bad status code: {r.status_code} returned by config url: {config_url}"
+                )
+        except Exception as e:
+            raise SemgrepError(
+                terminal_wrap(f"Failed to download config from {config_url}: {str(e)}")
+            )
+
+    @staticmethod
+    def _load_config_from_local_path(location: str) -> Dict[str, YamlTree]:
+        """
+        Return config file(s) as dictionary object
+        """
+        base_path = get_base_path()
+        loc = base_path.joinpath(location)
+        if loc.exists():
+            if loc.is_file():
+                return parse_config_at_path(loc)
+            elif loc.is_dir():
+                return parse_config_folder(loc)
+            else:
+                raise SemgrepError(f"config location `{loc}` is not a file or folder!")
+        else:
+            addendum = ""
+            if IN_DOCKER:
+                addendum = " (since you are running in docker, you cannot specify arbitrary paths on the host; they must be mounted into the container)"
+            raise SemgrepError(
+                terminal_wrap(
+                    f"Unable to find a config; path `{loc}` does not exist{addendum}"
+                )
+            )
+
+    def resolve_config(self, config_str: str) -> Dict[str, YamlTree]:
+        """ resolves if config arg is a registry entry, a url, or a file, folder, or loads from defaults if None"""
+        start_t = time.time()
+        extra_headers: Dict[str, str] = {}
+        if config_str == AUTO_CONFIG_KEY:
+            logger.warning(
+                terminal_wrap(
+                    "Auto config mode will use semgrep rules to scan your codebase and uses the Semgrep Registry"
+                    " to generate recommended rules based on your languages and frameworks."
+                    "\n"
+                ),
+            )
+            if self.project_url is not None:
+                extra_headers["X-Semgrep-Project"] = self.project_url
+                logger.warning(
+                    with_color(
+                        "green",
+                        terminal_wrap(
+                            f"Logging in to the Semgrep Registry as project '{self.project_url}'...\n"
```

Comment:
Can we just use `Fetching rules from semgrep registry for project : {self.project_url}` instead of logging in language

Reply:
cc. @ievans . This language is to communicate to the user that we're explicitly logging in with an identity.